### DISCUSSION TITLE
Unclear concerning what I should use for my noreply email

### DISCUSSION THREAD

**Discussion by rfalanga** [community/community#73393] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I've been a member of GitHub since 2015. I started to use the noreply email address for commits a few years after that, when I learned about the "Keep my email address private" feature. Once I saw that, I began to use the USERNAME@users.noreply.github.com email address when I configured a new repo locally.

At work I started a new GitHub account recently. I've done the same thing there, but when I went to get the noreply email address I saw ID+USERNAME@users.noreply.github.com syntax instead. So, I thought that's what I should use for that account.

Now, looking at the email settings in my older GitHub account I see the ID+USERNAME@users.noreply.github.com syntax. And looking at the [Setting your commit email address](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address) I am confused, because I see both there. Since my older account is created before 2015 do I use USERNAME@users.noreply.github.com or ID+USERNAME@users.noreply.github.com? And am I correct in thinking I should only use the ID+USERNAME@users.noreply.github.com syntax for my new, work GitHub account when configuring a new, local repo?

**Comment by airtower-luna** [comment, marked as answer]
As things are, you should use the old (`USERNAME@users.noreply.github.com`) address for the old account, and the ID-based one for the new account. There's one reason you might want to switch your old account to the ID-based format, [as described in the documentation](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-commit-email-addresses):

> If you use your `noreply` email address for GitHub to make commits and then [change your username](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/changing-your-github-username), those commits will not be associated with your account on GitHub.com. This does not apply if you're using the ID-based `noreply` address from GitHub. For more information, see "[Changing your GitHub username](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/changing-your-github-username)."

If you want to make the switch, the process is described in the blue notice box right above that paragraph.

**Reply by rfalanga** [reply]
Thank you for the explanation. That makes sense and I'll adopt what you've recommended, for both accounts.
